File: Easy/roman_to_integer/roman_to_integer.cpp
#include <iostream>
#include <vector>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
    public:
        int romanToInt(string s) {
            unordered_map<char, int> romanMap = {
                {'I', 1},
                {'V', 5},
                {'X', 10},
                {'L', 50},
                {'C', 100},
                {'D', 500},
                {'M', 1000}
            };

            int total = 0;
            for (size_t i = 0; i < s.length(); i++) {
                if (i + 1 < s.length() && romanMap[s[i]] < romanMap[s[i + 1]]) {
                    total -= romanMap[s[i]];
                } else {
                    total += romanMap[s[i]];
                }
            }
            return total;
        }
};

//Example usage:
int main() {
    Solution solution1;
    string roman1 = "III";
    int result1 = solution1.romanToInt(roman1);
    cout << "For the input \"III\", the integer value is: " << result1 << endl;

    Solution solution2;
    string roman2 = "LVIII";
    int result2 = solution2.romanToInt(roman2);
    cout << "For the input \"LVIII\", the integer value is: " << result2 << endl;

    Solution solution3;
    string roman3 = "MCMXCIV";
    int result3 = solution3.romanToInt(roman3);
    cout << "For the input \"MCMXCIV\", the integer value is: " << result3 << endl;
};

File: Easy/roman_to_integer/roman_to_integer.py
from typing import List

def romanToInt(s: str) -> int:
    values = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    }
    total = 0
    for i in range(len(s)):
        if (i+1 < len(s)) and (values[s[i]] < values[s[i+1]]):
            total -= values[s[i]]
        else:
            total += values[s[i]]
    return total

#Example usage: 
if __name__ == "__main__":
    print("Testing romanToInt function:")
    s1 = "III"
    print("For the input s1 'III', the integer value is:", romanToInt(s1)) #Output: 3

    s2 = "LVIII"
    print("For the input s2 'LVIII', the integer value is:", romanToInt(s2)) #Output: 58

    s3 = "MCMXCIV"
    print("For the input s3 'MCMXCIV', the integer value is:", romanToInt(s3)) #Output: 1994